openapi_python_client/templates/property_templates/union_property.pyi

```diff
@@ -2,22 +2,24 @@
 def _parse_{{ property.python_name }}(data: Any) -> {{ property.get_type_string() }}:
     data = None if isinstance(data, Unset) else data
     {{ property.python_name }}: {{ property.get_type_string() }}
-    {% for inner_property in property.inner_properties %}
-    {% if inner_property.template and not loop.last %}
+    {% for inner_property in property.inner_properties_with_template %}
+    {% if not loop.last or property.inner_properties_without_template %}
```

Comment:
This will loop through all the properties for each property with templates to recompute `inner_properties_without_template`. I generally prefer leaving methods that do work as methods instead of using `@property` to make it more obvious that "this does something".

I'll check this out and make a tweak to avoid any O(n^2) behavior which, while probably insignificant in most cases, I'd still like to avoid if possible.